File: src_cs/VirusBroadcast/PersonPool.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace VirusBroadcast {
	public class PersonPool {

		public static PersonPool Instance { get; } = new PersonPool();

		public List<Person> PersonList { get; } = new List<Person>();

		public int GetPeopleSize(Person.State state) {
			int i = 0;
			foreach (var person in PersonList) {
				if (person.CurState.CompareTo(state) == 0) {
					i++;
				}
			}
			return i;
		}

		public int GetPeopleSize() => PersonList.Count;

		public static int RECOVERED { get; set; } = 0;

		private PersonPool() {
			var city = new City(Constants.CITY_WIDTH, Constants.CITY_HEIGHT);
			// 添加城市居民
			for (var i = 0; i < Constants.POPULATION; i++) {
				var rand = new Random();
				int x = (int)rand.NextGaussian(100, city.CenterX);
				int y = (int)rand.NextGaussian(100, city.CenterY);
				if (x > Constants.CITY_WIDTH) {
					x = Constants.CITY_WIDTH;
				}
				if (x < -Constants.CITY_WIDTH) {
					x = -Constants.CITY_WIDTH;
				}
				if (y > Constants.CITY_HEIGHT) {
					y = Constants.CITY_HEIGHT;
				}
				if (y < -Constants.CITY_HEIGHT) {
					y = -Constants.CITY_HEIGHT;
				}
				PersonList.Add(new Person(city, x, y));
			}
		}
	}
}

File: VirusBroadcast_java_en/src/PersonPool.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

/**
 * Person Pool
 *
 */
public class PersonPool {
    private static PersonPool personPool = new PersonPool();

    public static PersonPool getInstance() {
        return personPool;
    }

    List<Person> personList = new ArrayList<Person>();

    public List<Person> getPersonList() {
        return personList;
    }


    /**
     * @param state Condition of people Person.State, null = total
     * @return Population of the certain condition of people
     */
    public int getPeopleSize(Person.State state) {
        if (state == null) {
            return personList.size();
        }
        int i = 0;
        for (Person person : personList) {
            if (person.getState().compareTo(state) == 0) {
                i++;
            }
        }
        return i;
    }

    public static int RECOVERED = 0; // Recovered times

    private PersonPool() {
        City city = new City(Constants.CITY_WIDTH / 2, Constants.CITY_HEIGHT / 2);
        // Add people
        for (int i = 0; i < Constants.POPULATION; i++) {
            Random random = new Random();
            //Generate number under N(a,b)：Math.sqrt(b)*random.nextGaussian()+a
            int x = (int) (100 * random.nextGaussian() + city.getCenterX());
            int y = (int) (100 * random.nextGaussian() + city.getCenterY());
            if (x > Constants.CITY_WIDTH) {
                x = Constants.CITY_WIDTH;
            }
            if (x < -Constants.CITY_WIDTH) {
                x = -Constants.CITY_WIDTH;
            }
            if (y > Constants.CITY_HEIGHT) {
                y = Constants.CITY_HEIGHT;
            }
            if (y < -Constants.CITY_HEIGHT) {
                y = -Constants.CITY_HEIGHT;
            }
            personList.add(new Person(city, x, y));
        }
    }
}
